dbms/src/Storages/MergeTree/MergeTreeDataWriter.cpp

```diff
@@ -146,8 +146,12 @@ MergeTreeData::MutableDataPartPtr MergeTreeDataWriter::writeTempPart(BlockWithDa
             ProfileEvents::increment(ProfileEvents::MergeTreeDataWriterBlocksAlreadySorted);
     }
 
+    auto compression_method = data.context.chooseCompressionMethod(
+        data.getTotalActiveSizeInBytes(),
+        static_cast<double>(data.getTotalActiveSizeInBytes()) / data.getTotalCompressedSize());
```

Comment:
This is little tricky. You need to provide data part size and ratio of data part size to all table size in arguments of `chooseCompressionMethod`. But data part size is not known in advance, before we write it.

This logic is intended for cases when existing data will be recompressed (as in case of merges and alters). And we take data part size to make decision, do we need to recompress it.
But when writing a new part, it is meaningless.

I think, you need to pass two zeros as arguments. Compression method will be selected only if both `min_part_size` and `min_part_size_ratio` are zeros or was not specified in configuration for corresponding compression method.

You definitely will specify `min_part_size` and `min_part_size_ratio` as zeros for choosing compression method `none`, because otherwise, `lz4` will be used.